plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/client/PinotClient.java

```diff
@@ -623,19 +643,38 @@ public static ResultsIterator fromResultTable(BrokerResponseNative brokerRespons
     }
 
     public static <T> T doWithRetries(int retries, Function<Integer, T> caller)
+    {
+        return doWithRetries(retries, caller, DEFAULT_RETRY_INTERVAL);
+    }
+
+    public static <T> T doWithRetries(int retries, Function<Integer, T> caller, int retryInterval)
```

Comment:
Why not using Failsafe?

Reply:
I'm not familiar with this lib. Feel free to modify this PR or point me to the sample usage.

Reply:
Here's an example usage: https://github.com/trinodb/trino/blob/master/plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/ReadSessionCreator.java#L111